File: SmartInventoryAPI/Models/Entities/Sales.cs
namespace SmartInventoryAPI.Models.Entities;

public class Sales
{
    public long ID { get; set; }
    public long Location_ID { get; set; }
    public int Sales_Status { get; set; }
    public DateTime Sales_Date { get; set; }
    public TimeSpan Sales_Time { get; set; }
    public bool Is_Deleted { get; set; } = false;
    public bool Is_Reserved { get; set; }
    public string? Sales_Number { get; set; }
    public long? Ref_Sales_Number { get; set; }
    public decimal Total_Amount { get; set; }
    //public DateTime Creation_Date { get; set; } = DateTime.UtcNow;

    // Foreign keys
    public virtual Location? Location { get; set; }
    public virtual Sales? RefSales { get; set; }
    public virtual ICollection<SalesItem>? Items { get; set; }
}

File: SmartInventoryFrontend/src/types/sales.ts
import { Entity } from './common'
import { Item } from './item'
import { Location } from './location'

export enum SalesStatus {
  Pending = 0,
  Completed = 1,
  Cancelled = 2,
}

export const SalesStatusLabel: Record<SalesStatus, string> = {
  [SalesStatus.Pending]: 'Pending',
  [SalesStatus.Completed]: 'Completed',
  [SalesStatus.Cancelled]: 'Cancelled',
}

export interface SalesItem {
  id: number
  item_ID: number
  item_Name?: string
  item_Code?: string
  item_Category?: string
  unit_Of_Measure?: string
  sold_Quantity: number
  sub_Total: number
  is_Promotion: boolean
  discount_Percentage: number
}

export interface Sales {
  id: number
  location_ID: number
  location_Name?: string
  sales_Status: SalesStatus
  sales_Date: string | Date
  sales_Time: string
  sales_Number?: string
  total_Amount: number
  creation_Date?: string | Date
  items?: SalesItem[]
}

export interface SalesFilters {
  skip: number
  take: number
  salesId?: number
  salesNumber?: string
  status?: SalesStatus
  dateFrom?: string
  dateTo?: string
}

export interface SalesState {
  sales: Sales[]
  currentSale: Sales | null
  loading: boolean
  error: string | null
  total: number
}
